### PR TITLE
BUG: .ix with PeriodIndex is fixed

### PR DESCRIPTION
**Pull request by max-sixty**
Solves https://github.com/pydata/pandas/issues/4125

It doesn't feel like an elegant solution, but I think it works. 
I think when Period gets its own dtype, these will all go away?

### CODE REVIEW

File: pandas/core/index.py [hunk 1]
```diff
@@ -2,18 +2,18 @@
...
+
+import numpy as np
 import pandas.tslib as tslib
 import pandas.lib as lib
 import pandas.algos as _algos
 import pandas.index as _index
 from pandas.lib import Timestamp, Timedelta, is_datetime_array
+
+from pandas.compat import range, zip, lrange, lzip, u, map
+from pandas import compat
```
Comment (maximilianr) [inline pandas/core/index.py]: This stuff is PyCharm - I can revert if you want

File: pandas/core/index.py [hunk 2]
```diff
@@ -973,7 +976,9 @@ def _convert_list_indexer(self, keyarr, kind=None):
         and we have a mixed index (e.g. number/labels). figure out
         the indexer. return None if we can't help
         """
-        if (kind is None or kind in ['iloc','ix']) and (is_integer_dtype(keyarr) and not self.is_floating()):
+        if (kind is None or kind in ['iloc', 'ix']) and (
+            is_integer_dtype(keyarr) and not self.is_floating() and not isinstance(keyarr, ABCPeriodIndex)):
```
Comment (jreback) [inline pandas/core/index.py]: use ``com.is_period_arraylike``

Comment (jreback) [inline pandas/core/index.py]: though you might simply be able to define ``_convert_list_indexer`` in ``PeriodIndex`` to handle this in a nicer way

Comment (maximilianr) [inline pandas/core/index.py]: Agree for most cases - when you have a `PeriodIndex` and you supply another as an index. 
But if you have an integer index and someone supplies a `PeriodIndex` as the key, I don't think you want that resolving to its `int`s. Does that make sense?

Would this be solved if we had a `Period` `dtype`? I couldn't find an open issue about that, but I think you've mentioned it a few times

Comment (jreback) [inline pandas/core/index.py]: period dtype would fix this but that's for the future

period right now is an integer dtype and that's the problem it's confused

File: pandas/core/index.py [hunk 3]
```diff
@@ -973,7 +976,9 @@ def _convert_list_indexer(self, keyarr, kind=None):
         and we have a mixed index (e.g. number/labels). figure out
         the indexer. return None if we can't help
         """
-        if (kind is None or kind in ['iloc','ix']) and (is_integer_dtype(keyarr) and not self.is_floating()):
+        if (kind is None or kind in ['iloc', 'ix']) and (
+            is_integer_dtype(keyarr) and not self.is_floating() and not com.is_period_arraylike(keyarr)):
+
```
Comment (jreback) [inline pandas/core/index.py]: on 2nd thought, this is going to cause a perf issue ,change this to ``isinstance(keyarr, ABCPeriodIndex)`` (you can import this at the top.

**Comment by jreback** [comment]
pls add a whatsnew note

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/879f5e90bea3b8c4849fd9f83a834e858dccd407

thanks!

(I changed this slightly in another commit). ``is_array_period_like`` does inference on the object and thus could be slow potentially

**Comment by maximilianr** [comment]
Cheers @jreback!

**Pull request closed without merging**